rcl_action/src/rcl_action/goal_state_machine.c

```diff
@@ -26,51 +26,40 @@ typedef rcl_action_goal_state_t
 rcl_action_goal_state_t
 _execute_event_handler(rcl_action_goal_state_t state, rcl_action_goal_event_t event)
 {
-  if (GOAL_STATE_ACCEPTED != state || GOAL_EVENT_EXECUTE != event) {
-    return GOAL_STATE_UNKNOWN;
-  }
+  assert(GOAL_STATE_ACCEPTED == state);
+  assert(GOAL_EVENT_EXECUTE == event);
   return GOAL_STATE_EXECUTING;
 }
 
 rcl_action_goal_state_t
 _cancel_goal_event_handler(rcl_action_goal_state_t state, rcl_action_goal_event_t event)
 {
-  if ((GOAL_STATE_ACCEPTED != state && GOAL_STATE_EXECUTING != state) ||
-    GOAL_EVENT_CANCEL_GOAL != event)
-  {
-    return GOAL_STATE_UNKNOWN;
-  }
+  assert(GOAL_STATE_ACCEPTED == state || GOAL_STATE_EXECUTING == state);
+  assert(GOAL_EVENT_CANCEL_GOAL == event);
```

Comment:
assert is compiled out in release mode, that's why the new warnings.

why a PR titled "Increase test coverage ..." is changing behavior?
If a behavior change was needed, that should've been part of a different PR (or at least, commits shouldn't have been squashed)